### ISSUE TITLE
Add build warning for scripts using `anon_X`

### ISSUE THREAD

**Issue by PeridexisErrant** [DFHack/dfhack#948]
This would be a nice warning to get in Travis, though it probably shouldn't break the build (yet).  Implementation might happen in the scripts repo once testing is set up there.

**Comment by lethosor** [comment]
Yeah, it looks like there are 7 different anon_X fields being used in the scripts repo right now (although at least one is just a fallback for one that was renamed - it's probably not necessary to maintain that one for compatibility any longer).

It would probably be better (and faster) to set this up in the scripts repo, yeah. I might also want to look into whether it's possible to require DFHack builds to have submodule build(s) passing too.

**Comment by PeridexisErrant** [comment]
Nice - I always enjoy fixing an issue and adding tests so it doesn't come back.

We could move all our script-related tests to the scripts repo, and use [travis-build](https://github.com/travis-ci/travis-build) to execute them, I think.  It would also be good to check that the docs build still works, but I'm not sure if this is possible without the main docs available (for cross-references etc) - some kind of linter should be possible though.

**Comment by lethosor** [comment]
I'm confused as to why travis-build would be necessary - it looks like it's what Travis uses internally to convert .travis.yml to a shell script and run it on VMs, so why not just a normal .travis.yml file?

And yeah, documentation checking would be hard. The best I can think of right now is keeping the existing checks in the DFHack repo and having basic ones run in the scripts repo too (the same would apply for the script syntax checks, since there are some lua/ruby files outside of the scripts folder).

**Comment by PeridexisErrant** [comment]
I was thinking that the scripts repo would have it's own travis setup, to test pull requests and so on, and then travis-build could be used to run `scripts/.travis.yml` from the DFHack build.

Agreed; some duplication of scripts/docs checking seems like the best option to me.

**Comment by lethosor** [comment]
Meant to get this done, but various things led to now being a good time to make a release (and this isn't part of release packages anyway).

**Comment by PeridexisErrant** [comment]
@mifki has been working on a lua tool that checks for missing fields:  [bay12](http://www.bay12forums.com/smf/index.php?topic=158675) / [github](https://github.com/mifki/df-luacheck).

It looks to me like an overlapping-but-separate issue, which would nicely complement a very simple grep-based ban on use of `anon_X` fields in the DFHack repos.

**Comment by PeridexisErrant** [comment]
```
#!/usr/bin/env python
import os, sys

status = 0
for root, _, files in os.walk('scripts'):
    for f in files:
        if f[-3:] in {'.rb', 'lua'}:
            thisfile = False
            p = os.path.join(root, f)
            with open(p) as fh:
                lines = fh.readlines()
            for i, l in enumerate(lines):
                if 'anon_' in l:
                    if not thisfile:
                        print('\n', p)
                        thisfile = True
                        status = 1
                    print(str(i) + ':\t' + l.strip())

if status != 0:
    print('\n\nPlease open an issue or pull request at\n'
          'https://github.com/DFHack/df-structures \n'
          'so unknown fields can be named and identified.')
    #sys.exit(1)
```

**Comment by lethosor** [comment]
Self-assigning this - I started work on travis for scripts a few weeks ago, although I didn't make much progress. We should definitely get this done by the next release, though.
